citystatmodel.cpp

```diff
@@ -180,28 +182,26 @@ bool CityStatModel::removeRows(int row, int count, const QModelIndex &parent)
     endRemoveRows();
 }
 
-void CityStatModel::saveDataInitiated()
+void CityStatModel::saveData()
 {
     unsigned int savedItemsCounter = 0;
-    bool isSavingSuccessfull = m_dataLoader->writeCityItemsList(m_cityItemsList, savedItemsCounter);
-    emit isSaveDataSuccessfull(isSavingSuccessfull, savedItemsCounter);
+
+    bool isSavingSuccessfull = m_dataLoader.writeCityItemsVector(m_cityItemsVector, savedItemsCounter);
+    emit dataSaved(isSavingSuccessfull, savedItemsCounter);
 }
 
-void CityStatModel::fillPopulationInfo()
+void CityStatModel::updatePopulationInfo()
 {
-    m_minPopulationElement = std::numeric_limits<unsigned int>::max();
-    m_maxPopulationElement = 0;
-    m_totalPopulationElement = 0;
-
-    for(int i = 0; i < m_cityItemsList.size(); i++)
-    {
-        if ( m_cityItemsList[i].cityPopulation > m_maxPopulationElement)
-            m_maxPopulationElement =  m_cityItemsList[i].cityPopulation;
+    auto result = std::minmax_element(m_cityItemsVector.begin(), m_cityItemsVector.end(),
+        [](CityItem city1, CityItem city2) {return city1.getPopulation() < city2.getPopulation();});
```

Comment:
instead of "CityItem city1" use "const auto &first/second", remember about copying